src/common/KeyboardAvoider.js

```diff
@@ -9,11 +9,35 @@ type Props = $ReadOnly<{|
   children: React$Node,
   style?: ViewStyleProp,
   contentContainerStyle?: ViewStyleProp,
+  keyboardVerticalOffset?: number,
 |}>;
 
 export default class KeyboardAvoider extends PureComponent<Props> {
   render() {
-    const { behavior, children, style, contentContainerStyle } = this.props;
+    const {
+      behavior,
+      children,
+      style,
+      contentContainerStyle,
+
+      // How much the top of `KeyboardAvoidingView`'s layout *parent*
+      // is displaced downward from the top of the screen.
+      //
+      // If this isn't set correctly, the keyboard will hide some of
+      // the bottom of the screen (an area whose height is what this
+      // value should have been set to).
+      //
+      // I think `KeyboardAvoidingView`'s implementation mistakes `x`
+      // and `y` from `View#onLayout` to be a `View`'s position
+      // relative to the top left of the screen. In reality, I'm
+      // pretty sure they represent a `View`'s position relative to
+      // its parent:
+      //   https://github.com/facebook/react-native-website/issues/2056#issuecomment-773618381
+      //
+      // But at least `KeyboardAvoidingView` exposes this prop, which
+      // we can use to balance the equation.
```

Comment:
> it looks like it's an implementation comment

It sort of is an implementation comment, in that it explains a weirdness in some code that it calls as part of doing its job, right? If (for some strange reason 😬) that weirdness were common or easily-obtained knowledge, the jsdoc might do OK to just say "passes some props straight through to `KeyboardAvoidingView`", I suppose? I guess if it were well-known, the implementation comment wouldn't be needed either, hmm.

I agree that this is easy to miss where it currently is, so it'll be smart to propagate it to the jsdoc where it'll be surer to be found.

Reply:
Yeah, as written it does both. But this contains some key information that the reader who's simply using this component will need to know -- so that information is part of the interface and belongs in the jsdoc.

I think one way to write this would be:
* Take the first two paragraphs and put them in jsdoc. Also s/KeyboardAvoidingView/KeyboardAvoider/ there.
  * I think that covers all the information needed for simply using this component.
* Then leave the remaining paragraphs formatted as an implementation comment -- they're really information that's not needed for using the component, but is there to explain how the implementation works for the sake of someone modifying or debugging it.

> If for some strange reason () that weirdness were common (or easily obtained) knowledge, the component's jsdoc might do OK to just say "passes some props straight through to `KeyboardAvoidingView`", I suppose? I guess if it were well-known, the implementation comment wouldn't be needed either, hmm.

Yep and yep :slightly_smiling_face: 

In other circumstances, where we're just passing some props through but they *are* reasonably well documented (or at any rate we don't have any thoughts to add to what's in their docs), it can indeed make sense for our interface to just say that:

https://github.com/zulip/zulip-mobile/blob/0735ff117baa22fee9f1ead1662d2c3d48951f4e/src/common/Input.js#L28-L42